### PR TITLE
Implement "Save Code to New File..." chat action

### PR DESCRIPTION
**Pull request by mkondratek**
Resolves https://linear.app/sourcegraph/issue/CODY-2801/implement-vscodewindowshowsavedialog-to-get-save-file-button

Cody part: https://github.com/sourcegraph/cody/pull/5184

## Test plan
1. Ask Cody chat to generate some code snippet
2. Save Code to New File...

## Demo
https://github.com/user-attachments/assets/c5b2d000-ad43-4a6f-80fc-524f8ef6f342

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -24,4 +24,4 @@ kotlin.stdlib.default.dependency=false
 nodeBinaries.commit=8755ae4c05fd476cd23f2972049111ba436c86d4
 nodeBinaries.version=v20.12.2
 cody.autocomplete.enableFormatting=true
-cody.commit=a7ba3c987888439299edb6c215bb8b3ed4765d1d
+cody.commit=7537429109d9eaf61da6daf49716abe169fbad6f
```
Comment (mkondratek) [inline gradle.properties]: - [x] TODO: update cody commit

File: src/integrationTest/kotlin/com/sourcegraph/cody/edit/DocumentCodeTest.kt [hunk 2]
```diff
@@ -12,13 +12,15 @@ import com.sourcegraph.cody.edit.widget.LensSpinner
...
 import com.sourcegraph.cody.util.CustomJunitClassRunner
+import kotlin.test.Ignore
 import org.hamcrest.Matchers.startsWith
 import org.junit.Assert.assertThat
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(CustomJunitClassRunner::class)
 class DocumentCodeTest : CodyIntegrationTextFixture() {
+  @Ignore
```
Comment (pkukielka) [inline src/integrationTest/kotlin/com/sourcegraph/cody/edit/DocumentCodeTest.kt]: What is the reason of that? Is it too flaky?

Comment (mkondratek) [reply src/integrationTest/kotlin/com/sourcegraph/cody/edit/DocumentCodeTest.kt]: yup

File: gradle.properties [hunk 3]
```diff
@@ -24,4 +24,4 @@ kotlin.stdlib.default.dependency=false
 nodeBinaries.commit=8755ae4c05fd476cd23f2972049111ba436c86d4
 nodeBinaries.version=v20.12.2
 cody.autocomplete.enableFormatting=true
-cody.commit=a7ba3c987888439299edb6c215bb8b3ed4765d1d
+cody.commit=93cb98ba48f935949af0af93c6219b1b7e9076ef
```
Comment (pkukielka) [inline gradle.properties]: I merged your other PR to Cody, I guess we need to update commit hash?

Comment (mkondratek) [reply gradle.properties]: yes

File: src/main/kotlin/com/sourcegraph/cody/edit/EditService.kt [hunk 4]
```diff
@@ -74,8 +70,11 @@ class EditService(val project: Project) {
       // TODO: We need to support the file-level operations.
       when (op) {
         is CreateFileOperation -> {
-          logger.warn("Workspace edit operation created a file: ${op.uri}")
-          return false
+          logger.info("Workspace edit operation created a file: ${op.uri}")
+          val file =
+              CodyEditorUtil.createFileOrScratchFromUntitled(project, op.uri, content = "")
+                  ?: return false
+          CodyEditorUtil.showDocument(project, file)
```
Comment (pkukielka) [inline src/main/kotlin/com/sourcegraph/cody/edit/EditService.kt:77]: Should create function really show a file? Isn't that a separate call from agent?

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt [hunk 5]
```diff
@@ -253,4 +260,21 @@ class CodyAgentClient(private val webview: NativeWebviewProvider) {
 
     logger.debug("webview/postMessage ${params.id}: ${params.message}")
   }
+
+  @JsonRequest("window/showSaveDialog")
+  fun window_showSaveDialog(): CompletableFuture<String> {
```
Comment (pkukielka) [inline src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt]: `showSaveDialog` in VSC have quite a few optional parameters which we should respect, if applicable (e.g. `title` and `defaultUri`)

```
	export interface SaveDialogOptions {
		/**
		 * The resource the dialog shows when opened.
		 */
		defaultUri?: Uri;

		/**
		 * A human-readable string for the save button.
		 */
		saveLabel?: string;

		/**
		 * A set of file filters that are used by the dialog. Each entry is a human-readable label,
		 * like "TypeScript", and an array of extensions, e.g.
		 * ```ts
		 * {
		 * 	'Images': ['png', 'jpg']
		 * 	'TypeScript': ['ts', 'tsx']
		 * }
		 * ```
		 */
		filters?: { [name: string]: string[] };

		/**
		 * Dialog title.
		 *
		 * This parameter might be ignored, as not all operating systems display a title on save dialogs
		 * (for example, macOS).
		 */
		title?: string;
	}
```

Comment (mkondratek) [reply src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt]: ~These options are always `undefined`~
image

---

whoops, probably you are right

**Review by pkukielka — APPROVED** [review]
Approved, but please fix the missing `window_showSaveDialog` parameter issue.

**Review by pkukielka — APPROVED** [review]
LGTM

**Pull request merged**